### PR TITLE
Pin Assign Opt in the Add Constraint file form

### PR DESCRIPTION
**Pull request by zaryabkhan-rs**
> ### Motivate of the pull request
> - [ ] To address an existing issue. 
> - [ ] Breaking new feature. If so, please describe details in the description part.

> ### Describe the technical details
> #### What is currently done? (Provide issue link if applicable)
>- Add pin assignment option in Add_Constraint_form dialog.

> #### What does this pull request change?
>  This PR improves in the following aspects:  
>- Pin Assignment Method Selection.

> 
> 

> ### Testing Steps:
>- create_design project_1
>- Select pin option from add_constraint_form and check on the placement setting either the selection is set or not
>- pin_loc_assign_method random/in_define_order  -->  change the pin assign selection from console.

> ### Which part of the code base require a change
> 
> - [ ] Library: <Specify the library name>
> - [ ] Plug-in: <Specify the plugin name>
> - [ ] Engine
> - [ ] Documentation
> - [ ] Regression tests
> - [ ] Continous Integration (CI) scripts

> ### Impact of the pull request

> - [ ] Added new options to GUI for Add Constraint Form

> ### Update Screenshots

image
image

### CODE REVIEW

File: src/Compiler/Compiler.h [hunk 1]
```diff
@@ -214,7 +214,7 @@ class Compiler {
   PackingOpt m_packingOpt = PackingOpt::None;
   GlobalPlacementOpt m_globalPlacementOpt = GlobalPlacementOpt::None;
   PlacementOpt m_placementOpt = PlacementOpt::None;
-  PinAssignOpt m_pinAssignOpt = PinAssignOpt::In_Define_Order;
+  PinAssignOpt m_pinAssignOpt;
```
Comment (KochynVolodymyr) [inline src/Compiler/Compiler.h]: I think default value should be assign

File: src/NewProject/add_constraints_form.cpp [hunk 2]
```diff
@@ -22,6 +26,13 @@ addConstraintsForm::addConstraintsForm(QWidget *parent)
 
   ui->m_ckkBoxCopy->setText(tr("Copy sources into project."));
   ui->m_ckkBoxCopy->setCheckState(Qt::CheckState::Checked);
+  ui->select_defineOrder->setChecked(true);
+  Compiler *compiler = GlobalSession->GetCompiler();
+  compiler->PinAssignOpts(Compiler::PinAssignOpt::In_Define_Order);
```
Comment (KochynVolodymyr) [inline src/NewProject/add_constraints_form.cpp]: shouldn't be here. But you need set ui->select_defineOrder or i->select_random depends on compiler->PinAssignOpts()

File: src/ProjNavigator/add_file_form.cpp [hunk 3]
```diff
@@ -19,6 +22,13 @@ AddFileForm::AddFileForm(QWidget *parent)
   box->setContentsMargins(0, 0, 0, 0);
   box->setSpacing(0);
   ui->m_frame->setLayout(box);
+  ui->select_defineOrder->setChecked(true);
+  Compiler *compiler = GlobalSession->GetCompiler();
+  compiler->PinAssignOpts(Compiler::PinAssignOpt::In_Define_Order);
```
Comment (KochynVolodymyr) [inline src/ProjNavigator/add_file_form.cpp]: same here

File: src/NewProject/add_constraints_form.cpp [hunk 4]
```diff
@@ -34,3 +45,11 @@ bool addConstraintsForm::IsCopySource() {
   return ui->m_ckkBoxCopy->checkState() == Qt::CheckState::Checked ? true
                                                                    : false;
 }
+
+void addConstraintsForm::pinAssign_opt_listen() {
+  Compiler *compiler = GlobalSession->GetCompiler();
+  if (!ui->select_random->isChecked())
```
Comment (KochynVolodymyr) [inline src/NewProject/add_constraints_form.cpp]: please change to if (ui->select_random->isChecked()). This is more ease to read.

File: src/ProjNavigator/add_file_form.cpp [hunk 5]
```diff
@@ -152,3 +162,11 @@ void AddFileForm::on_m_comboBoxSets_currentIndexChanged(const QString &arg1) {
   createdialog->deleteLater();
   m_widgetGrid->currentFileSet(ui->m_comboBoxSets->currentText());
 }
+
+void AddFileForm::pinAssign_opt_listen() {
+  Compiler *compiler = GlobalSession->GetCompiler();
+  if (!ui->select_random->isChecked())
```
Comment (KochynVolodymyr) [inline src/ProjNavigator/add_file_form.cpp]: same here

File: src/Compiler/Compiler.h (hunk 1 above)
Comment (zaryabkhan-rs) [reply src/Compiler/Compiler.h]: it reads the default value from the Add constraint form.

Comment (KochynVolodymyr) [reply src/Compiler/Compiler.h]: but it shouldn't

Comment (zaryabkhan-rs) [reply src/Compiler/Compiler.h]: Okay I will update

**Pull request closed without merging**